src/plugins/vis_type_timeseries/public/application/components/data_time_range_mode_label.tsx

```diff
@@ -0,0 +1,110 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import './data_time_range_mode_label.scss';
+
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiFlexItem, EuiText, EuiToolTip, EuiFlexGroup } from '@elastic/eui';
+import { getUISettings } from '../../services';
+import { convertIntervalIntoUnit, isAutoInterval, isGteInterval } from './lib/get_interval';
+import { createIntervalBasedFormatter } from './lib/create_interval_based_formatter';
+import { TIME_RANGE_DATA_MODES } from '../../../common/timerange_data_modes';
+import { PanelData } from '../../../common/types';
+
+const lastValueFormattedMessage = i18n.translate(
+  'visTypeTimeseries.dataTimeRangeModeLabel.lastValueMode',
+  {
+    defaultMessage: 'Last value',
+  }
+);
+
+interface DataTimeRangeModeLabelProps {
+  seriesData?: PanelData['data'];
+  panelInterval: number;
+  modelInterval: string;
+  modelTimeRangeMode: TIME_RANGE_DATA_MODES;
+}
+
+export const DataTimeRangeModeLabel = ({
+  seriesData,
+  panelInterval,
+  modelInterval,
+  modelTimeRangeMode,
+}: DataTimeRangeModeLabelProps) => {
+  const dateFormat = getUISettings().get('dateFormat');
+  const scaledDataFormat = getUISettings().get('dateFormat:scaled');
+
+  const getFormattedPanelInterval = () => {
+    const interval = convertIntervalIntoUnit(panelInterval, false);
+    return interval && `${interval.unitValue}${interval.unitString}`;
+  };
+
+  const getLastValueLabelWithTooltip = () => {
+    const formatter = createIntervalBasedFormatter(panelInterval, scaledDataFormat, dateFormat);
+    const lastBucketDate = formatter(seriesData![seriesData!.length - 1][0]);
+    const formattedPanelInterval =
+      (isAutoInterval(modelInterval) || isGteInterval(modelInterval)) &&
+      getFormattedPanelInterval();
+
+    const tooltipContent = (
+      <EuiFlexGroup direction="column" gutterSize="none">
+        <EuiFlexItem grow={false}>
+          <FormattedMessage
+            id="visTypeTimeseries.dataTimeRangeModeLabel.lastBucketDate"
+            defaultMessage="Bucket: {lastBucketDate}"
+            values={{ lastBucketDate }}
+          />
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <FormattedMessage
+            id="visTypeTimeseries.dataTimeRangeModeLabel.panelInterval"
+            defaultMessage="Interval: {formattedPanelInterval}"
+            values={{ formattedPanelInterval }}
+          />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+
+    return (
+      <EuiToolTip position="top" content={tooltipContent}>
+        <EuiText className="tvbLastValueLabel" color="subdued" size="xs">
+          {lastValueFormattedMessage}
+        </EuiText>
+      </EuiToolTip>
```

Comment:
Would be great to get an icon or some other indication that this is interactive, since it looks like plain text!

Reply:
An icon is a great idea actually